src/main/java/com/esri/samples/na/routing_around_barriers/RoutingAroundBarriersController.java

```diff
@@ -0,0 +1,335 @@
+          // check if there are stops to remove
+          if (!stopsList.isEmpty()) {
+
+            // remove the last stop from the stop list and the graphics overlay
+            stopsList.remove(stopsList.size() - 1);
+            stopsGraphicsOverlay.getGraphics().remove(stopsGraphicsOverlay.getGraphics().size() - 1);
+          }
+        } else if (btnAddBarrier.isSelected()) {
+          // check if there are barriers to remove
+          if (!barriersList.isEmpty()) {
+
+            // remove the last barrier from the barrier list and the graphics overlay
+            barriersList.remove(barriersList.size() - 1);
+            barriersGraphicsOverlay.getGraphics().remove(barriersGraphicsOverlay.getGraphics().size() - 1);
+          }
+        }
+      }
+    });
+  }
+
+  /**
+   * Uses the route task with the lists of stops and barriers to determine the route and display it.
+   */
+  @FXML
+  private void createRouteAndDisplay() {
+    if (stopsList.size() >= 2) {
+      // clear the previous route from the graphics overlay, if it exists
+      routeGraphicsOverlay.getGraphics().clear();
+
+      // add the existing stops and barriers to the route parameters
+      routeParameters.setStops(stopsList);
+      routeParameters.setPolygonBarriers(barriersList);
+
+      // apply the requested route finding parameters
+      routeParameters.setFindBestSequence(findBestSequenceCheckBox.isSelected());
+      routeParameters.setPreserveFirstStop(preserveFirstStopCheckBox.isSelected());
+      routeParameters.setPreserveLastStop(preserveLastStopCheckBox.isSelected());
+
+      // solve the route task
+      final ListenableFuture<RouteResult> routeResultFuture = routeTask.solveRouteAsync(routeParameters);
+      routeResultFuture.addDoneListener(() -> {
+        try {
+          RouteResult routeResult = routeResultFuture.get();
+          if (!routeResult.getRoutes().isEmpty()) {
+            // get the first route result
+            Route firstRoute = routeResult.getRoutes().get(0);
+
+            // create a geometry for this route
+            Geometry routeGeometry = firstRoute.getRouteGeometry();
+
+            // create a graphic for the route and add it to the graphics overlay
+            Graphic routeGraphic = new Graphic(routeGeometry, routeLineSymbol);
+            routeGraphicsOverlay.getGraphics().add(routeGraphic);
+
+            // retrieve route travel time, rounded to the nearest minute
+            String routeTravelTime = ((Long) Math.round(firstRoute.getTravelTime())).toString();
+
+            // retrieve route length (m), convert to km, and round to two decimal points
+            double routeLengthKm = firstRoute.getTotalLength() / 1000;
+            double routeLengthKmRounded = Math.round(routeLengthKm * 100d) / 100d;
+
+            // set the title of the TitledPane to display the information
+            String output = String.format("Route directions: %s min (%s mi)", routeTravelTime, routeLengthKmRounded);
```

Comment:
```java
String.format("Route directions: %d min (%.2f km)", Math.round(travelTime), travelLength / 1000)
```

Reply:
I'm guessing what you mean is to query the route travel time and length inline, and to show it in km?
In that case, I did:
`String.format("Route directions: %d min (%.2f km)", Math.round(firstRoute.getTravelTime()), firstRoute.getTotalLength() / 1000);`